### PR TITLE
Fix `WandbLogger.watch()` method annotation to accept None for the log parameter

### PR DESCRIPTION
**Pull request by daturkel**
This is a comically small change that fixed an annoyance I was having. The [wandb `watch` method](https://github.com/wandb/wandb/blob/f209a3e10208ce025146c1a5dacc668cf0143bcf/wandb/sdk/wandb_watch.py#L23) has a `log` argument which can be a string ("gradients", "parameters", "all") OR `None`. You might provide `None` if you want to call `watch` with `log_graph` but don't want to log the gradients or parameters. 

However, the WandBLogger `watch` method has a type annotation of `str` for the `log` parameter which is passed to the underlying `wandb.watch` `log` parameter. This PR simply changes that type annotation to `Optional[str]` so that `None` doesn't violate the type annotation.

The current [WandBLogger docs](https://lightning.ai/docs/pytorch/stable/extensions/generated/lightning.pytorch.loggers.WandbLogger.html) are kind of more "tutorial style" rather than a full reference of all the methods on the class, so there isn't necessarily a change that needs to be made. However, I'm open to suggestions.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--19237.org.readthedocs.build/en/19237/

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Comment by daturkel** [comment]
Thanks for the approvals!
